KeyFrame new_kf
--- 
KeyFrame kf